test/utils/components.js

```diff
@@ -37,3 +35,29 @@ export class ErrorBoundary extends React.Component {
     return this.props.children;
   }
 }
+
+/**
+ * Allows counting how many times the owner of `RenderCounter` rendered.
```

Comment:
Based on https://reactjs.org/docs/profiler.html#onrender-callback, would this be more accurate?

```suggestion
 * Allows counting how many times the owner of `RenderCounter` rendered or
 * a component within the RenderCounter tree "commits" an update.
```

Reply:
Yes! I didn't really think about this use case considering how we used `useEffect`. I'll think about how we better document this.